### DISCUSSION TITLE
Feature request: Add Speed Efficiency Graph

### DISCUSSION THREAD

**Discussion by mvaneijken** [teslamate-org/teslamate#2833] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Currently the efficiency dashboard only shows temperature efficiency. It would be great if that is enhanced showing speed efficiency.

### Expected Behavior

Show Wh/km usage for speed ranges, like:
100-105, 105-110, 110-115, etc. 

### Relevant log output

```shell
None
```

### Type of installation

Docker

### Version

1.27.1

**Comment by lq313c** [comment]
This would be cool, but not very easy. The API doesn't provide an instantaneous efficiency metric. Getting distance travelled over short timeframes (seconds) would be tricky because the odometer reading from the API is really coarse and imprecise. You'd have to calculate it from GPS position data. 

It would be a lot easier to make a histogram for power vs speed, although it would only be meaningful over large quantities of data.

**Comment by clement-nardi** [comment]
Hello,

I have created a new grafana dashboard on my teslamate instance which I think corresponds to your feature request.
Here is a screenshot:
<img width="576" alt="image" src="https://github.com/teslamate-org/teslamate/assets/5113912/18514b76-e0a6-45ae-83bd-b6dfddcf32cc">

And here is the dashboard:
[grafana_dashboard_teslamate_consumption_by_speed.json](https://github.com/teslamate-org/teslamate/files/13874119/grafana_dashboard_teslamate_consumption_by_speed.json)

Everything is based on the "power" and "speed" columns in the "positions" table.
The consumption at a given speed is based on the **average power** at a given speed, and I am not 100% satisfied by that. Indeed if you look at individual power values at a given speed, they vary a lot depending on the situation: accelerating, breaking, going uphill or downhill, wind, temperature, etc. 
I think it would have made more sense to use the median `PERCENTILE_CONT(0.5) WITHIN GROUP(ORDER BY power) as median_power` , but unfortunately the individual values are small integers which produces imprecise results.
Maybe another formula would make more sense, like the average of values between the 30th and 70th percentiles?
Still, averaging all values gives useful results.

I also hardcoded the battery capacity (56 kWh) in the theoretical range SQL query.

I'd love to see something like this in the efficiency dashboard!

Best regards,
Clément
